components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/StorageTxImpl.java

```diff
@@ -421,14 +421,14 @@ public BlobRef setBlob(final InputStream inputStream, final Map<String, String>
 
     final WritePolicy effectiveWritePolicy = writePolicySelector.select(asset, writePolicy);
     if (effectiveWritePolicy == WritePolicy.DENY) {
-      throw new IllegalOperationException("Repository is read only.");
+      throw new IllegalOperationException("Repository '" + getBucket().repositoryName() + "' is read only.");
```

Comment:
throw new IllegalOperationException("Repository is read only: " + getBucket().repositoryName()) is generally simpler and conforms better to how other exceptions are being created in the system.